### PR TITLE
[SL Beta 2] Add YNAB connector

### PR DESCRIPTION
**Pull request by RolandHewage**
Resolves https://github.com/wso2-enterprise/choreo/issues/7349

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

JDK 11
Ballerina Beta 2
Ubuntu 20.04

https://api.youneedabudget.com
https://api.youneedabudget.com/v1
https://app.youneedabudget.com
https://api.youneedabudget.com/papi/spec-v1-swagger.json
https://www.youneedabudget.com/zapier/

### CODE REVIEW

File: openapi/ynab/client.bal [hunk 1]
```diff
@@ -0,0 +1,401 @@
+import ballerina/http;
```
Comment (indikasampath2000) [inline openapi/ynab/client.bal:1]: license header missing

File: openapi/ynab/types.bal [hunk 2]
```diff
@@ -0,0 +1,604 @@
+public type PayeeLocation record {
```
Comment (indikasampath2000) [inline openapi/ynab/types.bal:1]: license header missing

File: openapi/ynab/client.bal (hunk 1 above)
Comment (RolandHewage) [reply openapi/ynab/client.bal:17]: Resolved & amended the changes. Thank you.

File: openapi/ynab/types.bal (hunk 2 above)
Comment (RolandHewage) [reply openapi/ynab/types.bal:17]: Resolved & amended the changes. Thank you.

**Review by indikasampath2000 — APPROVED** [review]

**Pull request merged**